converter.go

```diff
@@ -1022,6 +1003,153 @@ func arrowToValue(ctx context.Context, rowIdx int, srcColumnMeta execResponseRow
 	return nil, fmt.Errorf("unsupported data type")
 }
 
+func handleNativeArrowMap(ctx context.Context, rowIdx int, keyMetadata, valueMetadata fieldMetadata, srcValue arrow.Array, loc *time.Location, higherPrecision bool, params map[string]*string) (snowflakeValue, error) {
```

Comment:
i have impression we pass too many values between functions... perhaps we can introduce some kind of conversion context to wrap these that are inseparable?

Reply:
I was considering it, but each function has different arguments... I wouldn't introduce it. At the end I'd like to make some refactoring (if I found some time), maybe I will be able to clean it up a bit.

Reply:
Ok.